packages/perseus/src/types.js

```diff
@@ -75,6 +75,22 @@ export type Version = {|
 
 export type EditorMode = "edit" | "preview" | "json";
 
+export type ChoiceState = {|
+    selected: boolean,
+    crossedOut: boolean,
+    highlighted: boolean,
+    rationaleShown: boolean,
+    correctnessShown: boolean,
+    previouslyAnswered: boolean,
+    readOnly: boolean,
+|};
+
+export type RadioChoiceWithMetadata = {|
+    ...PerseusRadioChoice,
+    originalIndex: number,
+    correct: boolean,
+|};
```

Comment:
I really like this pattern, I think we should keep this in mind as we move forward.

Reply:
That's not as handy as overwriting them for sure, lol. Will make a note of this though. Thank you both!